packages/control-property-editor/src/components/ModeSwitcher.tsx

```diff
@@ -0,0 +1,44 @@
+import { setAppMode } from '@sap-ux-private/control-property-editor-common';
+import { UIDefaultButton, UILabel } from '@sap-ux/ui-components';
+import type { ReactElement } from 'react';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+import type { RootState } from '../store';
+import './ModeSwitcher.scss';
+
+/**
+ * React element for ModeSwitch.
+ *
+ * @returns ReactElement
+ */
+export function ModeSwitcher(): ReactElement {
+    const { t } = useTranslation();
+    const dispatch = useDispatch();
+
+    const mode = useSelector<RootState, 'navigation' | 'adaptation'>((state) => state.appMode);
+    const disabled = useSelector<RootState, boolean>((state) => state.isAppLoading);
+    return (
+        <div className="mode-switcher">
+            <UILabel>{t('MODE')}:</UILabel>
+            <UIDefaultButton
+                transparent={true}
+                checked={mode === 'adaptation'}
+                onClick={(): void => {
+                    dispatch(setAppMode('adaptation'));
+                }}
+                disabled={disabled}>
+                {t('EDIT')}
+            </UIDefaultButton>
+            <UIDefaultButton
+                transparent={true}
+                checked={mode === 'navigation'}
+                onClick={(): void => {
+                    dispatch(setAppMode('navigation'));
+                }}
+                disabled={disabled}>
+                {t('LIVE')}
+            </UIDefaultButton>
+        </div>
+    );
```

Comment:
I would suggest making two handlers here and wrapping them in useCallback for better performance:

```tsx
const handleAdaptationClick = useCallback(() => {
        dispatch(setAppMode('adaptation'));
    }, [dispatch]);

const handleNavigationClick = useCallback(() => {
    dispatch(setAppMode('navigation'));
}, [dispatch]);
```

And then use them accordingly:

```tsx
...
onClick={handleAdaptationClick}
...
```